### PR TITLE
Hds 335 add required indicator

### PR DESCRIPTION
**Pull request by KristinLBradley**
### :pushpin: Summary

Please add feedback on additional "required" indicators I added and let me know if I missed any. Thanks!

1. ~~Gather preliminary feedback & approval for my approach. ("Alert" used as 1st test case.)~~ DONE
2. Add "required" indicators within documentation of all components.

### :hammer_and_wrench: Detailed description

* Added markup for "required" indicator in docs for Alert component "type" argument.
* Added styles for "required" indicator.
* Grouped "required" indicator styles with existing similar "default" indicator styles.
* Updated gitignore to include VS Code workspace files.

### :camera_flash: Screenshots
**Notes:** I reused some existing styles from ".default" which seemed like a similar "indicator" type element. I used colors I found in the color design tokens which seemed the most appropriate. I added a border because the pale background color didn't seem to have enough contrast against the white background to me. Please feel free to suggest changes. :)

**UPDATED:**
Screen Shot 2022-07-20 at 11 51 59 AM

**OLD:**
Screen shot of proposed new 'required' indicator

***

### 👀 How to review

👉 Please add feedback on my initial approach and if you have suggestions for changes. (Visual look, code, etc.)

Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: packages/components/tests/dummy/app/styles/components/dummy-component-props.scss
Comment (KristinLBradley) [inline packages/components/tests/dummy/app/styles/components/dummy-component-props.scss:35]: code clean up

File: packages/components/tests/dummy/app/styles/components/dummy-component-props.scss
Comment (KristinLBradley) [inline packages/components/tests/dummy/app/styles/components/dummy-component-props.scss:61]: Moved default styles here so they could be grouped with the similar styles for "required" which I added. This changes the element the styles are scoped to which didn't seem to me that it would be a problem but let me know if I'm wrong.

File: packages/components/tests/dummy/app/templates/components/alert.hbs
Comment (KristinLBradley) [inline packages/components/tests/dummy/app/templates/components/alert.hbs:13]: Cleaned up formatting for consistency

File: packages/components/tests/dummy/app/templates/components/alert.hbs
Comment (KristinLBradley) [inline packages/components/tests/dummy/app/templates/components/alert.hbs:87]: I used a "strong" tag for the "required" indicator but let me know if you'd prefer to use a "span" or some other tag here.

Comment (MelSumner) [reply packages/components/tests/dummy/app/templates/components/alert.hbs]: I think `strong` is okay here, always good to have required things clearly marked for screen readers. 👍

**Review by MelSumner — APPROVED** [review]
I think this looks fine for the current site! I wonder how it would look if it was the same size as the type though? right now it looks a little ^_^....making it match the middle text would make it more ^-- (if that makes sense, sorry for the weird analogy)

File: packages/components/tests/dummy/app/templates/components/alert.hbs
Comment (KristinLBradley) [reply packages/components/tests/dummy/app/templates/components/alert.hbs]: Thanks!

**Comment by KristinLBradley** [comment]
> 

@MelSumner To me it looked a little weird having the "required" use the larger font size but that could be just me.

**PRed:** smaller font size for "required" text (matching size of "default" indicator):
Screen Shot 2022-07-19 at 1 54 22 PM

**NOT PRed:** Bigger font size for "required" matching "type" text fontsize:
git pull origin main
Screen Shot 2022-07-19 at 3 05 49 PM

**Comment by MelSumner** [comment]
Oh so you already did what I was thinking! Funny how our eyes play tricks on us. I have no other issues!

File: packages/components/tests/dummy/app/styles/components/dummy-component-props.scss
Comment (didoo) [inline packages/components/tests/dummy/app/styles/components/dummy-component-props.scss:72]: we don't use design tokens (and components) from the HDS design system to build the UI of the documentation, to avoid coupling the two and making things harder for us.
it's OK if you just want to use the corresponding hex color values, though.

File: packages/components/tests/dummy/app/styles/components/dummy-component-props.scss
Comment (didoo) [inline packages/components/tests/dummy/app/styles/components/dummy-component-props.scss:69]: I think this is a leftover

File: packages/components/tests/dummy/app/styles/components/dummy-component-props.scss
Comment (didoo) [reply packages/components/tests/dummy/app/styles/components/dummy-component-props.scss:73]: yes, it's ok because we may want to fine tune the size to have optical balance with the rest of the text.

In that sense, I have noticed that, despite having a smaller font size, optically the "required" tag seems bigger than the text next to it. 
screenshot_1673
and the reason is that the font-family is different. I suggest to use the same font family as the normal `<code>` element so visually they're more balanced:
screenshot_1674

(and potentially reduce the vertical padding for the required, or risks to become too prominent)

**Review by didoo — COMMENTED** [review]
@KristinLBradley added a few comments, and set up a meeting for later, so we can discuss in person. overall, all works for me, just some fine tunings.

**Comment by didoo** [comment]
> Funny how our eyes play tricks on us

@MelSumner see comment here: https://github.com/hashicorp/design-system/pull/467#discussion_r925326870
they're not tricking you, it's really bigger (visually) and that's because of the different font-family between the two

File: packages/components/tests/dummy/app/styles/components/dummy-component-props.scss
Comment (KristinLBradley) [reply packages/components/tests/dummy/app/styles/components/dummy-component-props.scss:72]: Thanks, I'll update the styles of the "required" tag to use the code element's default font.

Note: The HTML code element typically uses a mono-spaced font to differentiate text which is meant to represent code from other "normal" text content. I agree it does look more balanced visually to use the same font for the "required" tag where I included it though. (We may want to use a different layout position or design solution to represent "required" values instead of using a mono-spaced font but we could look at that more in a future "non-scrappy" doc site.)

File: packages/components/tests/dummy/app/styles/components/dummy-component-props.scss
Comment (KristinLBradley) [reply packages/components/tests/dummy/app/styles/components/dummy-component-props.scss:61]: Thanks, I'll update it to repeat the shared styles.

Commentary: In general, I usually try to standardize visual styling of similar elements where it makes sense and try to avoid duplicating code. Typically, I would either separate out the common styles so that they could be referenced by both elements or keep what seem to be similar items like this grouped. Then if a need arose in the future to visually differentiate them I would refactor at that time.

File: packages/components/tests/dummy/app/styles/components/dummy-component-props.scss
Comment (KristinLBradley) [reply packages/components/tests/dummy/app/styles/components/dummy-component-props.scss]: After updating the font-family I deleted the font-size I had set so that it would inherit the size from the immediate parent.

**Comment by Dhaulagiri** [comment]
@KristinLBradley if you rebase your PR against `main` (or otherwise merge it in) this will go away. It looks like a change that was already merged there addresses this issue.

**Comment by KristinLBradley** [comment]
@didoo Any more feedback on this one? If it looks ok now then I can proceed with updating the docs for other components to add "required" where needed.

Btw, would it be better to merge this PR (after approval) & then start a new PR for other changes or would it be ok to use this same PR for all the updates to component docs? (I'm fine with using this same PR but let me know if you think it would make the PR too long or confusing.)

**Review by didoo — APPROVED** [review]
Looks good. Feel free to merge, or continue the work in the same PR.

File: packages/components/tests/dummy/app/templates/components/button.hbs
Comment (KristinLBradley) [inline packages/components/tests/dummy/app/templates/components/button.hbs:67]: I added 2 "default" tags here for clarity but let me know if you want me to undo these changes for this PR.

**Comment by KristinLBradley** [comment]
@didoo @MelSumner 

Please re-review. I added "required" indicators for the other component docs. Let me know if I missed any.

File: packages/components/tests/dummy/app/templates/components/link/inline.hbs
Comment (didoo) [inline packages/components/tests/dummy/app/templates/components/link/inline.hbs:80]: [nitpick] this technically speaking is not required. it's better if it's provided, of course, but is not `required` per se

File: packages/components/tests/dummy/app/templates/components/link/standalone.hbs
Comment (didoo) [inline packages/components/tests/dummy/app/templates/components/link/standalone.hbs:117]: [issue/blocker] this is definitely not required. it may be generated via a `@route` argument, see https://design-system-components-hashicorp.vercel.app/components/link/standalone#:~:text=URLs%20and%20routes%20handling

File: packages/components/tests/dummy/app/templates/components/dropdown.hbs
Comment (didoo) [inline packages/components/tests/dummy/app/templates/components/dropdown.hbs:157]: [issue] there are other props in the different dropdown sub-components that probably need a `required` tag

**Review by didoo — CHANGES_REQUESTED** [review]

**Comment by KristinLBradley** [comment]
@didoo Pushed new updates from feedback. Please re-check.

**Review by didoo — APPROVED** [review]
👍

**Pull request merged**
